modin/pandas/general.py

```diff
@@ -404,26 +404,34 @@ def concat(
                 sort=sort,
             )
         )
-    if join not in ["inner", "outer"]:
+    if join == "outer":
+        # Filter out empties
```

Comment:
Why does this filtering need to be done at the modin level of the dataframe, and not at the hdk backend level?

Reply:
The filtering was already implemented here - https://github.com/modin-project/modin/pull/5673/files#diff-270ead8b42be5edcb8f1979a86ef8cbc663d116fb7f0fa14e51999ded69e6c08L422 . I've changed the logic a bit to be compatible with Pandas.

Reply:
It is at least strange that filtering is now only for `outer` type of join. What about `inner` join?

Reply:
> It is at least strange that filtering is now only for `outer` type of join. What about `inner` join?

It's incorrect to filter out empty frames in case of inner join. In this case, we have to return an empty frame, but preserve the index.

Reply:
So you have found another bug that is present for all backends. We need to at least create a separate issue and add tests not only for hdk.

Reply:
> So you have found another bug that is present for all backends. We need to at least create a separate issue and add tests not only for hdk.

What about moving the test here https://github.com/modin-project/modin/blob/master/modin/pandas/test/test_concat.py ?

Reply:
If these tests reproduce the problem with `inner` concat (for example, for Ray engine and pandas backend) and it is possible to reduce the amount of hdk-specific code, then why not?